cmd/sonobuoy/app/pluginCache.go

```diff
@@ -0,0 +1,266 @@
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deletePlugin(getPluginCacheLocation(cmd), filenameFromArg(args[0], ".yaml"))
+		},
+	}
+
+	cmd.AddCommand(listCmd, showCmd, installCmd, uninstallCmd)
+
+	return cmd
+}
+
+// getPluginCacheLocation will return the location of the plugin cache (defaults to ~/.sonobuoy)
+// If no override is set in the env, the default is assumed. If we are unable to expand to an abosolute
+// path then we return the empty string signalling the feature should be disabled.
+func getPluginCacheLocation(cmd *cobra.Command) string {
+	usePath := os.Getenv(SonobuoyDirEnvKey)
+	if len(usePath) == 0 {
+		usePath = defaultSonobuoyDir
+	}
+	expandedPath, err := homedir.Expand(usePath)
+	if err != nil {
+		logrus.Errorf("failed to expand sonobuoy directory %q: %v", usePath, err)
+		return ""
+	}
+
+	if _, err := os.Stat(expandedPath); err != nil && os.IsNotExist(err) {
+		logrus.Debugf("sonobuoy plugin location %q does not exist, creating it.", expandedPath)
+		if err := os.Mkdir(expandedPath, 0777); err != nil {
+			logrus.Errorf("failed to create directory for installed plugins %q: %v", expandedPath, err)
+		}
+	}
+
+	logrus.Debugf("Using plugin cache location %q", expandedPath)
+	return expandedPath
+}
+
+func listInstalledPlugins(installedDir string) error {
+	pluginFiles, err := loadPlugins(installedDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to load installed plugins")
+	}
+
+	prefix := ""
+	first := true
+	for filename, p := range pluginFiles {
+		if !first {
+			prefix = "---\n"
+		}
+		fmt.Printf("%vfilename: %v\nplugin name: %v\nsource URL: %v\ndescription: %v\n",
+			prefix, filename, p.SonobuoyConfig.PluginName, p.SonobuoyConfig.SourceURL, p.SonobuoyConfig.Description)
+		first = false
+	}
+
+	return nil
+}
+
+// loadPlugins loads all plugins from the given directory (recursively) and
+// returns a map of their filename and contents. Returns the first error encountered
+// but continues to load and return as many plugins as possible.
+func loadPlugins(installedDir string) (map[string]*manifest.Manifest, error) {
+	pluginMap := map[string]*manifest.Manifest{}
+	var firstErr error
+
+	err := filepath.Walk(installedDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && filepath.Ext(path) == ".yaml" {
+			m, err := loader.LoadDefinitionFromFile(path)
+			if err != nil {
+				fmt.Printf("failed to load path %q: %v\n", path, err)
+				return errors.Wrapf(err, "failed to load definition from file %q", path)
+			}
+			pluginMap[path] = m
+		}
+		return nil
+	})
+	if err != nil {
+		if firstErr == nil {
+			firstErr = err
+		}
+		logrus.Errorf("error walking the path %q: %v\n", installedDir, err)
+	}
+
+	return pluginMap, firstErr
+}
+
+func loadPlugin(installedDir, reqFile string) (*manifest.Manifest, error) {
```

Comment:
Could this be done from the pre-loaded `map[string]*manifest.Manifest` from above ?

Reply:
I thought about doing it that way too; it could definitely leverage it if we wanted but the main difference was simply that this exits early if found. That's not a huge issue since users won't have 1 million plugins or anything. You're probably right that the simplification is preferred.